tests/tests_lite/utilities/test_data.py

```diff
@@ -11,13 +14,19 @@
     _get_dataloader_init_args_and_kwargs,
     _replace_dunder_methods,
     _replace_value_in_saved_args,
+    _unpatch_dunder_methods,
     _update_dataloader,
     _WrapAttrTag,
     has_iterable_dataset,
     has_len,
 )
 from lightning_lite.utilities.exceptions import MisconfigurationException
 
+# The classes already get patched on import of LightningLite
+# We need to unpatch first to have a clean starting point for all assertions in the tests below
+_unpatch_dunder_methods(DataLoader)
+_unpatch_dunder_methods(BatchSampler)
```

Comment:
I believe this will run just once when the test files are imported (during test collection), not before the tests just in this file run.

I think the safest option would be to patch and unpatch through a conftest autouse fixture